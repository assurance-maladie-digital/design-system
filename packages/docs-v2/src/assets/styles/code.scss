// Reset Vuetify styles & tweak theme
.v-application .nuxt-content-highlight,
.v-application .doc-code-block {
	pre {
		padding: 0;
		border-radius: 4px;
		background-color: #1f1f1f;
	}

	pre[class*=language] {
		&::after {
			content: attr(data-lang);
			bottom: .75rem;
			color: #bdbdbd;
			font-family: inherit;
			font-size: .7rem;
			font-weight: 700;
			position: absolute;
			right: 1rem;
			text-transform: uppercase;
		}
	}

	code {
		font-family: inherit;
		font-size: 1em;
		font-weight: 400;
		background-color: transparent;
		color: inherit;
		padding: 0;
	}

	pre,
	code {
		text-shadow: none;
	}

	code * {
		font-family: inherit;
	}
}

.v-application .nuxt-content-highlight pre {
	padding: 12px 16px;
}
