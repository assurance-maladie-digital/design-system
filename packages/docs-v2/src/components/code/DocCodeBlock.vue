<template>
	<VSheet
		:tile="tile"
		color="#1f1f1f"
		class="doc-code-block overflow-hidden"
		:class="{ 'mb-4': !noMargin }"
		outlined
		rounded
		dark
	>
		<slot />

		<CopyBtn
			:tooltip-duration="1000"
			:text-to-copy="getSlotContent"
			:vuetify-options="copyBtnOptions"
			label="Copier le code"
		/>
	</VSheet>
</template>

<script lang="ts">
	import Vue from 'vue';
	import Component, { mixins } from 'vue-class-component';

	import { mdiContentCopy } from '@mdi/js';

	const Props = Vue.extend({
		props: {
			tile: {
				type: Boolean,
				default: false
			},
			noMargin: {
				type: Boolean,
				default: false
			}
		}
	});

	const MixinsDeclaration = mixins(Props);

	@Component
	export default class DocCodeBlock extends MixinsDeclaration {
		copyIcon = mdiContentCopy;

		copyBtnOptions = {
			menu: {
				left: true,
				nudgeLeft: 16
			},
			icon: {
				color: 'grey lighten-1'
			}
		};

		getSlotContent(): string {
			const markup = this.$el.querySelector('pre');

			if (!markup) {
				return '';
			}

			return markup.innerText;
		}
	}
</script>

<style lang="scss">
	.doc-code-block {
		position: relative;
		padding: 12px 52px 12px 16px;

		pre {
			margin: 0;
		}

		.vd-copy-btn {
			position: absolute;
			background-color: inherit;
			top: 6px;
			right: 8px;
		}

		&:not(:hover) .vd-copy-btn .v-btn {
			opacity: .6;
		}

		/**
		 * prism.js tomorrow night eighties for JavaScript, CoffeeScript, CSS and HTML
		 * Based on https://github.com/chriskempson/tomorrow-theme
		 * @author Rose Pritchard
		 */
		code[class*="language-"],
		pre[class*="language-"] {
			color: #ccc;
			background: none;
			font-family: Consolas, Monaco, 'Andale Mono', 'Ubuntu Mono', monospace;
			font-size: 1em;
			text-align: left;
			white-space: pre;
			word-spacing: normal;
			word-break: normal;
			word-wrap: normal;
			line-height: 1.5;
			tab-size: 4;
			hyphens: none;
		}

		/* Code blocks */
		pre[class*="language-"] {
			overflow: auto;
		}

		/* Inline code */
		:not(pre) > code[class*="language-"] {
			padding: .1em;
			border-radius: .3em;
			white-space: normal;
		}

		.token.comment,
		.token.block-comment,
		.token.prolog,
		.token.doctype,
		.token.cdata {
			color: #999;
		}

		.token.punctuation {
			color: #ccc;
		}

		.token.tag,
		.token.attr-name,
		.token.namespace,
		.token.deleted {
			color: #e2777a;
		}

		.token.function-name {
			color: #6196cc;
		}

		.token.boolean,
		.token.number,
		.token.function {
			color: #f08d49;
		}

		.token.property,
		.token.class-name,
		.token.constant,
		.token.symbol {
			color: #f8c555;
		}

		.token.selector,
		.token.important,
		.token.atrule,
		.token.keyword,
		.token.builtin {
			color: #cc99cd;
		}

		.token.string,
		.token.char,
		.token.attr-value,
		.token.regex,
		.token.variable {
			color: #7ec699;
		}

		.token.operator,
		.token.entity,
		.token.url {
			color: #67cdcc;
		}

		.token.important,
		.token.bold {
			font-weight: bold;
		}

		.token.italic {
			font-style: italic;
		}

		.token.entity {
			cursor: help;
		}

		.token.inserted {
			color: green;
		}
	}
</style>
