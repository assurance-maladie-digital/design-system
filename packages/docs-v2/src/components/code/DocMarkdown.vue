<script land="ts">
	import Vue from 'vue';

	import remark from 'remark';
	import html from 'remark-html';

	const remarkProcessor = remark().use(html);

	function markdownToHtml(content) {
		return remarkProcessor.processSync(content);
	}

	export default Vue.extend({
		name: 'DocMarkdown',
		functional: true,
		props: {
			tag: {
				type: String,
				default: 'div'
			}
		},
		render(h, {
			children: nodes = [],
			data,
			props,
			slots
		}) {
			const children = [];
			const node = nodes[0] || {};

			if (node.children) {
				children.push(...node.children);
			} else if (nodes.length > 1) {
				children.push(nodes);
			} else {
				const text = node.text || data.domProps?.textContent || '';

				data.domProps = {
					...data.domProps,
					innerHTML: markdownToHtml(text)
				};
			}

			data.staticClass = `v-markdown ${data.staticClass || ''}`.trim();

			return h(props.tag, data, children);
		}
	});
</script>

<style lang="scss" scoped>
	.v-markdown ::v-deep p {
		margin-bottom: 0 !important;
	}
</style>
