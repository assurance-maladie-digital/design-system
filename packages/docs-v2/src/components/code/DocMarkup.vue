<template>
	<DocCodeBlock v-bind="$attrs">
		<Prism
			:code="code"
			:inline="inline"
			:language="language"
			:data-lang="dataLang"
		/>
	</DocCodeBlock>
</template>

<script lang="ts">
	import Vue from 'vue';
	import Component, { mixins } from 'vue-class-component';

	import Prism from 'vue-prism-component';

	const Props = Vue.extend({
		props: {
			language: {
				type: String,
				default: undefined
			},
			inline: {
				type: Boolean,
				default: false
			},
			code: {
				type: String,
				default: undefined
			},
			hideLanguage: {
				type: Boolean,
				default: false
			}
		}
	});

	const MixinsDeclaration = mixins(Props);

	@Component({
		components: {
			Prism
		},
		inheritAttrs: false
	})
	export default class DocMarkup extends MixinsDeclaration {
		get dataLang(): string | undefined {
			if (this.hideLanguage) {
				return;
			}

			return this.language;
		}
	}
</script>
