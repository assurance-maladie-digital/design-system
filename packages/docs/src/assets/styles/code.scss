// Reset Vuetify styles & tweak theme
.v-application {
	.nuxt-content-highlight,
	.doc-code-block {
		pre {
			padding: 0;
			background: #020d1f;
		}

		pre[class*=language] {
			&::after {
				content: attr(data-lang);
				bottom: .75rem;
				color: #bdbdbd;
				font-family: inherit;
				font-size: .7rem;
				font-weight: 700;
				position: absolute;
				right: 1rem;
				text-transform: uppercase;
			}
		}

		code {
			font-family: inherit;
			font-size: 1em;
			font-weight: 400;
			background-color: transparent;
			color: inherit;
			padding: 0;
		}

		pre,
		code {
			text-shadow: none;
		}
	}

	.nuxt-content-highlight pre {
		padding: 12px 16px;
	}

	span.token {
		font-family: inherit;
		background-color: transparent !important;
	}

	.token.selector,
	.token.attr-name,
	.token.string,
	.token.char,
	.token.builtin,
	.token.inserted {
		color: #008972;
	}

	.token.atrule,
	.token.attr-value,
	.token.keyword {
		color: #0084b2;
	}

	.token.operator,
	.token.entity,
	.token.url,
	.language-css .token.string,
	.style .token.string {
		color: #906b15;
	}

	.token.property,
	.token.tag,
	.token.boolean,
	.token.number,
	.token.constant,
	.token.symbol,
	.token.deleted {
		color: #87104d;
	}

	.language-css .token.string {
		color: #c08f1c;
	}
}
