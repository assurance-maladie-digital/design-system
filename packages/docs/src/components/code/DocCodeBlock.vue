<template>
	<VSheet
		:tile="tile"
		:rounded="rounded"
		:color="sheetColor"
		:class="{ 'mb-4': !noMargin }"
		class="doc-code-block overflow-hidden"
		outlined
		dark
	>
		<slot />

		<CopyBtn
			:tooltip-duration="1000"
			:text-to-copy="getSlotContent"
			:vuetify-options="copyBtnOptions"
			label="Copier le code"
		/>
	</VSheet>
</template>

<script lang="ts">
	import Vue from 'vue';
	import Component, { mixins } from 'vue-class-component';

	import { mdiContentCopy } from '@mdi/js';

	import { tokens } from '@cnamts/design-tokens';

	import '~/assets/styles/code-block-theme.scss';

	const Props = Vue.extend({
		props: {
			tile: {
				type: Boolean,
				default: false
			},
			rounded: {
				type: Boolean,
				default: true
			},
			noMargin: {
				type: Boolean,
				default: false
			}
		}
	});

	const MixinsDeclaration = mixins(Props);

	@Component
	export default class DocCodeBlock extends MixinsDeclaration {
		copyIcon = mdiContentCopy;

		copyBtnOptions = {
			menu: {
				left: true,
				nudgeLeft: 16
			},
			icon: {
				color: 'grey lighten-1'
			}
		};

		sheetColor = tokens.colors.amBlue.darken80;

		getSlotContent(): string {
			const markup = this.$el.querySelector('pre');

			if (!markup) {
				return '';
			}

			return markup.innerText;
		}
	}
</script>

<style lang="scss" scoped>
	.doc-code-block {
		width: 100%;
		position: relative;
		padding: 12px 52px 12px 16px;

		:deep() {
			pre[class*="language-"] {
				margin: 0;
			}

			.vd-copy-btn {
				position: absolute;
				background-color: inherit;
				top: 6px;
				right: 8px;
			}

			&:not(:hover) .vd-copy-btn .v-btn {
				opacity: .6;
			}
		}
	}
</style>
