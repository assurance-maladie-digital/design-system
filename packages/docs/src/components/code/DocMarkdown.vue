<script lang="ts">
	import Vue, { VNode } from 'vue';

	import { remark } from 'remark';
	import html from 'remark-html';

	import { VFile } from 'vfile';

	const remarkProcessor = remark().use(html, {
		// Disable sanitization since we trust the source
		// and raw HTML in Markdown cannot be sanitized
		sanitize: false
	});

	function markdownToHtml(content: string): VFile {
		return remarkProcessor.processSync(content) as unknown as VFile;
	}

	export default Vue.extend({
		name: 'DocMarkdown',
		functional: true,
		props: {
			tag: {
				type: String,
				default: 'div'
			}
		},
		render(h, {
			children: nodes = [],
			data,
			props
		}): VNode | VNode[] {
			const children = [];
			const node = nodes[0] || {};

			if (node.children) {
				children.push(...node.children);
			} else if (nodes.length > 1) {
				children.push(nodes);
			} else {
				const text: string = node.text || data.domProps?.textContent || '';

				data.domProps = {
					...data.domProps,
					innerHTML: markdownToHtml(text.trim())
				};
			}

			data.staticClass = `v-markdown ${data.staticClass || ''}`.trim();

			return h(props.tag, data, children);
		}
	});
</script>

<style lang="scss" scoped>
	.v-markdown ::v-deep p {
		margin-bottom: 0 !important;
	}
</style>
