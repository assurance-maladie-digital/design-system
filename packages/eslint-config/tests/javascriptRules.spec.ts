import { ESLint } from 'eslint';

import { javascript } from './fixtures/code';

/* eslint-disable-next-line @typescript-eslint/no-var-requires */
const baseConfig = require('../javascript');

describe('JavaScript configuration', () => {
	const eslint = new ESLint({
		useEslintrc: false,
		baseConfig
	});

	it('lints the code correctly', async() => {
		const results = await eslint.lintText(javascript);

		expect(results[0].errorCount).toEqual(6);
	});

	it('applies the correct rules', async() => {
		const results = await eslint.lintText(javascript);

		expect(results[0].messages).toMatchSnapshot();
	});
});
