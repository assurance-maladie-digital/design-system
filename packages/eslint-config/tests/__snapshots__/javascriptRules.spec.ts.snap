// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JavaScript configuration applies the correct rules 1`] = `
Array [
  Object {
    "column": 1,
    "endColumn": 12,
    "endLine": 1,
    "fix": Object {
      "range": Array [
        0,
        3,
      ],
      "text": "let",
    },
    "line": 1,
    "message": "Unexpected var, use let or const instead.",
    "messageId": "unexpectedVar",
    "nodeType": "VariableDeclaration",
    "ruleId": "no-var",
    "severity": 2,
  },
  Object {
    "column": 12,
    "endColumn": 1,
    "endLine": 2,
    "fix": Object {
      "range": Array [
        11,
        11,
      ],
      "text": ";",
    },
    "line": 1,
    "message": "Missing semicolon.",
    "messageId": "missingSemi",
    "nodeType": "VariableDeclaration",
    "ruleId": "semi",
    "severity": 2,
  },
  Object {
    "column": 1,
    "endColumn": 25,
    "endLine": 2,
    "fix": Object {
      "range": Array [
        12,
        15,
      ],
      "text": "let",
    },
    "line": 2,
    "message": "Unexpected var, use let or const instead.",
    "messageId": "unexpectedVar",
    "nodeType": "VariableDeclaration",
    "ruleId": "no-var",
    "severity": 2,
  },
  Object {
    "column": 19,
    "endColumn": 20,
    "endLine": 2,
    "fix": Object {
      "range": Array [
        30,
        31,
      ],
      "text": "",
    },
    "line": 2,
    "message": "Unexpected space before function parentheses.",
    "messageId": "unexpectedSpace",
    "nodeType": "FunctionExpression",
    "ruleId": "space-before-function-paren",
    "severity": 2,
  },
  Object {
    "column": 25,
    "endColumn": 1,
    "endLine": 3,
    "fix": Object {
      "range": Array [
        36,
        36,
      ],
      "text": ";",
    },
    "line": 2,
    "message": "Missing semicolon.",
    "messageId": "missingSemi",
    "nodeType": "VariableDeclaration",
    "ruleId": "semi",
    "severity": 2,
  },
  Object {
    "column": 9,
    "endColumn": 1,
    "endLine": 4,
    "fix": Object {
      "range": Array [
        45,
        45,
      ],
      "text": ";",
    },
    "line": 3,
    "message": "Missing semicolon.",
    "messageId": "missingSemi",
    "nodeType": "ExpressionStatement",
    "ruleId": "semi",
    "severity": 2,
  },
]
`;
